### PR TITLE
fix valid commit dict

### PR DESCRIPTION
**Pull request by ipa-hsd**
FIxes some errors for "What is the size of the change difference?" in the Jupyter notebook.   

Note: some bug files have "'bad fix hash" errors when the commit is not part of a PR:
```
skipping bug [missing stats]: BugDescription(filename='/home/ipa-hsd/projects/rosin/robust/care-o-bot/ca7b536/ca7b536.bug')
skipping bug [missing stats]: BugDescription(filename='/home/ipa-hsd/projects/rosin/robust/geometry2/15b2e3c/15b2e3c.bug')
skipping bug [missing stats]: BugDescription(filename='/home/ipa-hsd/projects/rosin/robust/care-o-bot/43705f7/43705f7.bug')
skipping bug [missing stats]: BugDescription(filename='/home/ipa-hsd/projects/rosin/robust/universal_robot/89145c4/89145c4.bug')
skipping bug [missing stats]: BugDescription(filename='/home/ipa-hsd/projects/rosin/robust/geometry2/566092b/566092b.bug')
skipping bug [missing stats]: BugDescription(filename='/home/ipa-hsd/projects/rosin/robust/other/473ab5e/473ab5e.bug')
skipping bug [missing stats]: BugDescription(filename='/home/ipa-hsd/projects/rosin/robust/geometry2/e12e723/e12e723.bug')
skipping bug [missing stats]: BugDescription(filename='/home/ipa-hsd/projects/rosin/robust/geometry2/0c3aef3/0c3aef3.bug')
skipping bug [missing stats]: BugDescription(filename='/home/ipa-hsd/projects/rosin/robust/turtlebot/a4f35ee/a4f35ee.bug')
```
Also errors for `confidential` bugs since no repos.

---

Edit: Changes made in the PR:
- generate commit statistics for the Jupyter notebook section "What is the size of the change difference?":
  
  - fix `fix` commit hashes and PRs (rationale added to bug descriptions): https://github.com/robust-rosin/robust/pull/399/commits/39724ce863423b51efc3fbb96d3a4be90a2681c3, https://github.com/robust-rosin/robust/pull/399/commits/7bda2ba214530a86e413f694cd5d60f75429da96, https://github.com/robust-rosin/robust/pull/399/commits/bf3595cca94a74aa0039be70a3156d5df09c4585
  - fix "no forks exist": https://github.com/robust-rosin/robust/pull/399/commits/7195752474515828bfb9df9b49e83502669bd4ec, https://github.com/robust-rosin/robust/pull/399/commits/f4d6b7e8c610b9f2b239dc6a6a589da55617b39e  
    additionally, also created forks on `robust-rosin` when missing.
  - Identify bugs that were not fixed and set `fix` field to `null`: https://github.com/robust-rosin/robust/pull/399/commits/73cd29f6df8c740f5a85f0f45efb7c4b3f24275c, https://github.com/robust-rosin/robust/pull/399/commits/885c016bd18700fe4098cd93bdc72935579af4b0
  - Identify descriptions that are not bugs and add an identifier file `NOT_A_BUG`: https://github.com/robust-rosin/robust/pull/399/commits/85a8ba17dedfaade1ee79870583d9041a406518b, https://github.com/robust-rosin/robust/pull/399/commits/f5d14bd19f4c742855e92de1fc450011920ac99e 
  - Finally, generate commit stats from the script `fetch-commit-data`: https://github.com/robust-rosin/robust/pull/399/commits/bab581ca5aa8bfeb6ff5a035ced156ec128ba0d3
 
- Some additional changes not related to this PR:
  - Removed `license` field from bug description: https://github.com/robust-rosin/robust/pull/399/commits/78dd598ddd240f0e0e0b96ff7a0f01e2abd6efbd   
    The field does not have any influence on the nature of the bug and subsequently will not influence this study.
  - Correct fault-failure codes based on the discussions in https://github.com/robust-rosin/robust/issues/408 and https://github.com/robust-rosin/robust/issues/407: https://github.com/robust-rosin/robust/pull/399/commits/0b0c8cd1f28e6958829b416d83a0a88c40e6c85f

### CODE REVIEW

File: motoman/045ab54/045ab54.bug [hunk 1]
```diff
@@ -28,8 +28,20 @@ bug:
   reproducibility: always
   trace: N/A
 fix:
-  repo: https://github.com/ros-industrial/motoman
-  hash: 045ab5403733ee9ec2512ee4d22e541dd3d6424f
+  commits:
+    - repo: https://github.com/ros-industrial/motoman
```
Comment (ChrisTimperley) [inline motoman/045ab54/045ab54.bug:32]: This looks malformed?

**Comment by ChrisTimperley** [comment]
What's the status on this one? It would be great to get this into `master` so we can run the notebook from there.

**Comment by ipa-hsd** [comment]
`bad fix hash`:  
[geometry2/0c3aef3/0c3aef3.bug](https://github.com/robust-rosin/robust/blob/master/geometry2/0c3aef3/0c3aef3.bug#L62):  [This](https://github.com/ros/geometry2/issues/275) ticket addresses the issue. The correct hash is [this](https://github.com/pjreed/novatel_gps_driver/commit/9c9220f6e8a98473d7b25793fe764eace474af80)    
[universal_robot/89145c4/89145c4.bug](https://github.com/robust-rosin/robust/blob/3d3142dec28ba51ad435f14a4f749e764e622224/universal_robot/89145c4/89145c4.bug#L55): Was finally fixed in https://github.com/ros-industrial/universal_robot/pull/268 in https://github.com/ros-industrial/universal_robot/commit/2e078f5e3e9b3ec52dac83c096e90e7cd3edd0f2  
[geometry2/566092b/566092b.bug](https://github.com/robust-rosin/robust/blob/master/geometry2/566092b/566092b.bug): the fork was created (2018-04-09) before the commit  
[geometry2/15b2e3c/15b2e3c.bug](https://github.com/robust-rosin/robust/blob/3d3142dec28ba51ad435f14a4f749e764e622224/geometry2/15b2e3c/15b2e3c.bug#L48): just bad format  
[care-o-bot/ca7b536/ca7b536.bug](https://github.com/robust-rosin/robust/blob/3d3142dec28ba51ad435f14a4f749e764e622224/care-o-bot/ca7b536/ca7b536.bug#L50): correct one seems to be https://github.com/ipa320/cob_control/pull/47/commits/47adcee14c7f39645352f2226b32c5f116ea0c22
[geometry2/e12e723/e12e723.bug](https://github.com/robust-rosin/robust/blob/3d3142dec28ba51ad435f14a4f749e764e622224/geometry2/e12e723/e12e723.bug#L55): does not exist. Correct one is mentioned [here](https://github.com/robust-rosin/robust/blob/master/geometry2/e12e723/e12e723.bug#L65)  
  
bugs in `other` have `URL` https://git.code.tecnalia.com, which are private repos. `confidential` have no hash mentioned.

**Comment by ipa-hsd** [comment]
[care-o-bot/43705f7/43705f7.bug](https://github.com/robust-rosin/robust/blob/3d3142dec28ba51ad435f14a4f749e764e622224/care-o-bot/43705f7/43705f7.bug#L46): the PR was closed. Correct one in https://github.com/ipa320/cob_common/pull/183/commits/499dbb637b6acaade0daa0f36acac798b7dd34d2

**Comment by ipa-hsd** [comment]
@ChrisTimperley it should be good to be merged. Could you please review it?

**Review by ChrisTimperley — APPROVED** [review]
As far as I can tell by scanning through the files, this looks good.

**Comment by gavanderhoorn** [comment]
This PR seems to remove the `license` field: was that discussed somewhere?

What's the rationale?

**Comment by gavanderhoorn** [comment]
(commenting here as I cannot comment on file moves)

For visibility/discoverability it might be better to create a `not-a-bug` directory in the directories for each of the systems-examined, instead of replicating the directory structure of the `systems-examined` in a central `not-a-bug` directory at root level.

**Comment by ipa-hsd** [comment]
Changes made in the PR:
- generate commit statistics for the Jupyter notebook section "What is the size of the change difference?":
  
  - fix `fix` commit hashes and PRs (rationale added to bug descriptions): https://github.com/robust-rosin/robust/pull/399/commits/39724ce863423b51efc3fbb96d3a4be90a2681c3, https://github.com/robust-rosin/robust/pull/399/commits/7bda2ba214530a86e413f694cd5d60f75429da96, https://github.com/robust-rosin/robust/pull/399/commits/bf3595cca94a74aa0039be70a3156d5df09c4585
  - fix "no forks exist": https://github.com/robust-rosin/robust/pull/399/commits/7195752474515828bfb9df9b49e83502669bd4ec, https://github.com/robust-rosin/robust/pull/399/commits/f4d6b7e8c610b9f2b239dc6a6a589da55617b39e  
    additionally, also created forks on `robust-rosin` when missing.
  - Identify bugs that were not fixed and set `fix` field to `null`: https://github.com/robust-rosin/robust/pull/399/commits/73cd29f6df8c740f5a85f0f45efb7c4b3f24275c, https://github.com/robust-rosin/robust/pull/399/commits/885c016bd18700fe4098cd93bdc72935579af4b0
  - Identify descriptions that are not bugs and add an identifier file `NOT_A_BUG`: https://github.com/robust-rosin/robust/pull/399/commits/85a8ba17dedfaade1ee79870583d9041a406518b, https://github.com/robust-rosin/robust/pull/399/commits/f5d14bd19f4c742855e92de1fc450011920ac99e 
  - Finally, generate commit stats from the script `fetch-commit-data`: https://github.com/robust-rosin/robust/pull/399/commits/bab581ca5aa8bfeb6ff5a035ced156ec128ba0d3
 
- Some additional changes not related to this PR:
  - Removed `license` field from bug description: https://github.com/robust-rosin/robust/pull/399/commits/78dd598ddd240f0e0e0b96ff7a0f01e2abd6efbd   
    The field does not have any influence on the nature of the bug and subsequently will not influence this study.
  - Correct fault-failure codes based on the discussions in https://github.com/robust-rosin/robust/issues/408 and https://github.com/robust-rosin/robust/issues/407: https://github.com/robust-rosin/robust/pull/399/commits/0b0c8cd1f28e6958829b416d83a0a88c40e6c85f

**Comment by ipa-hsd** [comment]
@ChrisTimperley @gavanderhoorn @git-afsantos @wasowski

**Review by gavanderhoorn — APPROVED** [review]

**Comment by gavanderhoorn** [comment]
I've copied https://github.com/robust-rosin/robust/pull/399#issuecomment-829912436 to the PR description.

**Comment by ipa-hsd** [comment]
> Edit: (Additional) changes made in the PR:

The 1st section ("generate commit statistics for the Jupyter notebook") was the purpose of the PR (referring to the "(Additional)" )

**Comment by gavanderhoorn** [comment]
> > Edit: (Additional) changes made in the PR:
> 
> The 1st section ("generate commit statistics for the Jupyter notebook") was the purpose of the PR (referring to the "(Additional)" )

please feel free to update the PR description.

I merely felt 'hiding' it in a comment somewhere down in the timeline was not the best place for such information.

**Comment by ipa-hsd** [comment]
Thanks for your comments @ChrisTimperley and @gavanderhoorn.

**Pull request merged**